### ISSUE TITLE
block_on + Timer not work

### ISSUE THREAD

**Issue by ansjsun** [stjepang/smol#153]
it is ok .
````
smol::run(async {
       Timer::after(Duration::from_secs(1)).await;
 });
````

bu when use 
````
smol::block_on(async {
        Timer::after(Duration::from_secs(1)).await;
    });
````

it will hang

**Comment by stjepang** [comment]
This is by design in the current version of smol, but I will fix this in smol 0.2

**Comment by ansjsun** [comment]
Look forward to
